feat: Konnect SSO reference (#959)
Kong/developer.konghq.com: app/konnect-sso.md

      * [Azure group filtering](https://learn.microsoft.com/en-us/azure/active-directory/hybrid/connect/how-to-connect-fed-group-claims#group-filtering) 
      * [Okta group filtering](https://support.okta.com/help/s/article/How-to-send-certain-groups-that-the-user-is-assigned-to-in-one-Group-attribute-statement)

      You may need to contact the support team of your identity provider in order to learn how to filter groups emitted for the application.
---

{{site.konnect_short_name}} supports external single sign-on authentication using an Identity Provider (IdP). Using SSO in {{site.konnect_short_name}}, you can enable authentication for the following:
* **The {{site.konnect_short_name}} platform:** Allow [Org admins](/teams-and-roles/) to log in with SSO. This is an alternative to {{site.konnect_short_name}}'s [built-in authentication](https://cloud.konghq.com/global/organization/settings#authentication-scheme).
* **Dev Portals:** Allow developers to log in to the [Dev Portal](/dev-portal/) with SSO.

SSO for each of these is configured through different settings, so enabling one doesn't automatically enable the other. Both methods support OIDC and SAML-based SSO.

{:.warning}

> > We recommend using a single authentication method, however, {{site.konnect_short_name}} supports the ability to
> combine built-in authentication with _either_ OIDC or SAML based SSO. Combining both OIDC and SAML based SSO is not supported.
> Keep built-in authentication enabled while you are testing IdP authentication. Only disable built-in authentication after
> successfully testing the configurations in these guides.

Reviewer: ```suggestion
> We recommend using a single authentication method, however, {{site.konnect_short_name}} supports the ability to combine built-in authentication with _either_ OIDC or SAML based SSO. Combining both OIDC and SAML based SSO is not supported.
Keep built-in authentication enabled while you are testing IdP authentication. Only disable built-in authentication after successfully testing the configurations in these guides.
```

## Supported IdP providers

> While technically any OIDC or SAML-compliant provider can be used, the following have been verified:

Reviewer: ```suggestion
{{site.konnect_short_name}} supports any OIDC or SAML-compliant provider. The following have been verified to work out-of-the-box:
```

* Okta 
* Azure Active Directory
* Oracle Identity Cloud Service 
* Keycloak

## SSO configuration

> Before you configure SSO in {{site.konnect_short_name}}, you must configure the following in your IdP:

Reviewer: ```suggestion
To configure SSO in {{site.konnect_short_name}}, you must configure the following in your IdP:
```

> * {{site.konnect_short_name}} must be added to your IdP as an application

Reviewer: ```suggestion
* Add {{site.konnect_short_name}} to your IdP as an application
```

> * Users that need to use SSO are added to your IdP tenant

Reviewer: ```suggestion
* Add users that need to use SSO to the IdP tenant.
```

* Claims are set up in your IdP

You can configure {{site.konnect_short_name}} SSO in the following ways:

<!--vale off-->
{% table %}
columns:
  - title: Feature
    key: feature
  - title: UI setting
    key: ui
  - title: API endpoint
    key: api
rows:
  - feature: "{{site.konnect_short_name}} platform"
    ui: "Go to the [Authentication Scheme organization settings](https://cloud.konghq.com/global/organization/settings#authentication-scheme)<sup>1</sup>"
    api: "[`/identity-providers`](/api/konnect/identity/v3/#/operations/create-identity-provider)<sup>1</sup>"
  - feature: "Dev Portal"
    ui: "Go to the Identity settings for your [Dev Portal](https://cloud.konghq.com/portals/)"
    api: "[`/portals/{portalId}/identity-providers`](/api/konnect/portal-management/v2/#/operations/create-portal-identity-provider)"

{% endtable %}
<!--vale on-->

{:.info}

> > **1:** When you configure the organization login path, enter a unique string that will be used in the URL your users use to log in. For example: `examplepath`.

Reviewer: ```suggestion
> **note:** When you configure the organization login path, enter a unique string that will be used in the URL your users use to log in. For example: `examplepath`.
```

    ui: "Go to the [Team Mapping](https://cloud.konghq.com/global/organization/settings#team-mappings ) in the Organization settings."
    api: "[`/identity-provider/team-mappings`](/api/konnect/identity/v3/#/operations/update-idp-team-mappings)"
  - feature: "Dev Portal"
    ui: "Click on a Dev Portal](https://cloud.konghq.com/portals/) and go to Team Mappings in its settings."
    api: "[`/portals/{portalId}/identity-provider/team-group-mappings`](/api/konnect/portal-management/v2/#/operations/update-portal-team-group-mappings)"

{% endtable %}
<!--vale on-->

When you configure team mappings for the {{site.konnect_short_name}} org, keep the following in mind:
* Each {{site.konnect_short_name}} team can be mapped to **one** IdP group. You must have at least one group mapped to save configuration changes.
* To manage user and team memberships in {{site.konnect_short_name}} from the Organization settings, select the **Konnect Mapping Enabled** checkbox.
* To assign team memberships by the IdP during SSO login via group claims mapped to {{site.konnect_short_name}} teams, select the **IdP Mapping Enabled** checkbox and enter your IdP groups in the relevant fields.

{:.warning}

> > **Important:** While it is not required, we recommend using {{site.konnect_short_name}}’s group to team mapping feature. If you choose not to use this feature, then approving new users will require a two step process. First, the user will need to login to {{site.konnect_short_name}} with their IdP credentials. They will receive an access error, but the new user will be visible to the {{site.konnect_short_name}} administrator. The administrator can now map the user to a valid {{site.konnect_short_name}} team, which will give the user the required access. The new user must now re-login to gain access.

Reviewer: This note feels wrong, can we remove it?
